src/mesh/analysis.rs

```diff
@@ -222,9 +265,27 @@ pub fn border_edge_loops(edge_sharing: &EdgeSharingMap) -> Vec<Vec<OrientedEdge>
                     break;
                 }
 
+                let other_edge_reverted = other_edge.to_reverted();
+                if other_edge_reverted.chains_to(front) {
+                    current_chain.push_front(other_edge_reverted);
+                    border_edges.remove(i);
+                    found = true;
+
+                    break;
+                }
+
                 let back = current_chain[current_chain.len() - 1];
                 if back.chains_to(*other_edge) {
-                    current_chain.push_back(border_edges.remove(i));
+                    current_chain.push_back(*other_edge);
+                    found = true;
```

Comment:
For symmetry and therefore readability, `found` should be last, as is in the other blocks.